app/models/web_hook_events_daily_aggregate.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+class WebHookEventsDailyAggregate < ActiveRecord::Base
+  belongs_to :web_hook
+
+  default_scope { order("created_at DESC") }
+  after_create :aggregate
+
+  def self.purge_old
+    where("created_at < ?", SiteSetting.retain_web_hook_events_period_days.days.ago).delete_all
+  end
+
+  def self.by_day(start_date, end_date, web_hook_id = nil)
+    result = where("date >= ? AND date <= ?", start_date.to_date, end_date.to_date)
+    result = result.where(web_hook_id: web_hook_id) if web_hook_id
+    result
+  end
+
+  def aggregate
+    events =
+      WebHookEvent.where(
+        "created_at >= ? AND created_at < ? AND web_hook_id = ?",
+        self.date,
+        self.date + 1.day,
+        self.web_hook_id,
+      )
+
+    self.mean_duration = events.sum(:duration) / events.count if events.count > 0
```

Comment:
```suggestion
    self.mean_duration = events.sum(:duration) / events.count if events.positive?
```

Reply:
Actually this check shouldn't really be needed; if there hasn't been any events for the day, we should not create this WebHookEventsDailyAggregate record